backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InitializersLowering.kt

```diff
@@ -83,8 +83,7 @@ internal class InitializersLowering(val context: CommonBackendContext) : ClassLo
 
                     // We shall keep initializer for constants for compile-time instantiation.
                     declaration.initializer =
-                            if (initExpression is IrConst<*> &&
-                                    (initExpression.type.isPrimitiveType() || initExpression.type.isString())) {
+                            if (initExpression is IrConst<*>) {
```

Comment:
What exactly is this required for?

Reply:
https://github.com/JetBrains/kotlin/blob/b81ec2c8f05df39cfb7918d7c2032f0a1c8c6df9/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/optimizations/PropertyAccessorInlineLowering.kt#L51

Reply:
So this code expects that initializer of `const` property is `IrConst`. This is probably true, but the expectation still makes this code fragile. Additionally checking for `property.isConst` should make this code more reliable.